packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1645,15 +1648,26 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     // Check if the new value is the same as the current local value, or is the same
     // as the post-formatting value of the previous pass.
     final bool textChanged = _value?.text != value?.text;
-    final bool isRepeat = value?.text == _lastFormattedUnmodifiedTextEditingValue?.text;
-    if (textChanged && !isRepeat && widget.inputFormatters != null && widget.inputFormatters.isNotEmpty) {
-      for (final TextInputFormatter formatter in widget.inputFormatters)
+    final bool isRepeatText = value?.text == _lastFormattedUnmodifiedTextEditingValue?.text;
+    final bool isRepeatSelection = value?.selection == _lastFormattedUnmodifiedTextEditingValue?.selection;
```

Comment:
Is it ok if either of these is true in a `null == null` situation?

Reply:
Yeah, the `.text` and `.selection` values are asserted to be non-null (so the ? is probably unnecessary) and if the entire value passed is null, then that is a case of repeat that we want to filter as we would already know what the formatted result is.